File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Sliding window Technique/6-1456. Maximum Number of Vowels in a Substring of Given Length/main.ts
function maxVowels(s: string, k: number): number {
  let vowels = new Set(["a", "e", "i", "o", "u"]);
  let count = 0;
  for (let i = 0; i < k; i++) count += Number(vowels.has(s[i]));
  let ans = count;
  for (let i = k; i < s.length; i++) {
    count += Number(vowels.has(s[i])) - Number(vowels.has(s[i - k]));
    ans = Math.max(ans, count);
  }

  return ans;
}

File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Sliding window Technique/6-1456. Maximum Number of Vowels in a Substring of Given Length/main.java
class Solution {
    public int maxVowels(String s, int k) {
        Set<Character> vowels = Set.of('a', 'i', 'o', 'u', 'e');
        int count = 0;
        for (int i = 0; i < k; i++)
            count += vowels.contains(s.charAt(i)) ? 1 : 0;
        int ans = count;
        for (int i = k; i < s.length(); i++) {
            count += vowels.contains(s.charAt(i)) ? 1 : 0;
            count -= vowels.contains(s.charAt(i - k)) ? 1 : 0;
            ans = Math.max(count, ans);
        }
        return ans;
    }
}